### PR TITLE
Fix double emails for webshop payments (and refactor payment related objects)

### PR DESCRIPTION
**Pull request by oleast**
## What kind of a pull request is this?

- QA / Code Review

## Code Checklist

- [x] The code follows dotkom code style 
- [x] The code passes the defined tests
- [ ] I have added tests for the code I added
- [ ] I have provided documentation for the code I added
- [x] The code is ready to be merged

## (Possible) Breaking Changes

- [x] The changes are backwards compatible

## Description of changes

Payments for webshop have sent two receipt emails previously because the OrderLine sent a receipt and the payment relation sent its own receipt.
This makes payments handle related objects/models with an abstract interface to gather needed information about the payment.

This also solves a lot of problems with circular imports which where a result of the _is_type checking

### CODE REVIEW

File: apps/payment/mixins.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+from abc import abstractmethod
+from typing import List
+
+from django.conf import settings
+from django.utils import timezone
+
+from apps.authentication.models import OnlineUser as User
+
+
+class PaymentMixin:
```
Comment (henrikhorluck) [inline apps/payment/mixins.py:10]:
```suggestion
class PaymentMixin(ABC):
```
This is an abstract class

File: apps/payment/mixins.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+from abc import abstractmethod
```
Comment (henrikhorluck) [inline apps/payment/mixins.py:1]:
```suggestion
from abc import ABC, abstractmethod
```
Se below

File: apps/payment/mixins.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        Handle the effects of payment relation being refunded.
+        """
+        pass
+
+    @abstractmethod
+    def get_payment_receipt_items(self, payment_relation) -> List[dict]:
+        return []
+
+
+class ReceiptMixin:
```
Comment (henrikhorluck) [inline apps/payment/mixins.py:60]:
```suggestion
class ReceiptMixin(ABC):
```

**Comment by oleast** [comment]
> Great! While you are here, it would be great if you added orderings to the models, so that we can slowly, but surely reduce the amounts of warnings we get in CI

I would like to just focus on the issues at hand, and not everything else related to it. That because _way_ too much of a rabbit hole

File: apps/payment/mixins.py (hunk 1 above)
Comment (oleast) [reply apps/payment/mixins.py:10]: I tried this out, and it just created a clusterfuck with the metaclasses in migrations. I'll leave it out for now.

**Pull request merged**